### PR TITLE
add task solution

### PR DESCRIPTION
**Pull request by AndriiYelieva**

### CODE REVIEW

File: src/figuresClasses.ts [hunk 1]
```diff
@@ -1,19 +1,75 @@
...
+    const s = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
 
+    return Math.floor(area * 100) / 100;
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  shape: string;
```
Comment (etojeDenys) [inline src/figuresClasses.ts:33]: Initialize shape directly with the value: shape: string = 'circle';

File: src/figuresClasses.ts [hunk 2]
```diff
@@ -1,19 +1,75 @@
...
+
+  constructor(public color: string, public radius: number) {
+    if (radius <= 0) {
+      throw new Error('Radius must be greater than 0');
+    }
+
+    this.color = color;
```
Comment (etojeDenys) [inline src/figuresClasses.ts:40]: Remove this line as the color is already assigned in the constructor parameters

File: src/figuresClasses.ts [hunk 3]
```diff
@@ -1,19 +1,75 @@
...
+    this.shape = 'circle';
+    this.radius = radius;
```
Comment (etojeDenys) [inline src/figuresClasses.ts:42]: Remove this line as the radius is already assigned in the constructor parameters

File: src/figuresClasses.ts [hunk 4]
```diff
@@ -1,19 +1,75 @@
...
+  }
 
+  getArea(): number {
+    return Math.floor(Math.PI * this.radius ** 2 * 100) / 100;
+  }
 }
 
-export function getInfo(figure) {
+export class Rectangle implements Figure {
+  shape: string;
```
Comment (etojeDenys) [inline src/figuresClasses.ts:51]: Initialize shape directly with the value: shape: string = 'rectangle';

File: src/figuresClasses.ts [hunk 5]
```diff
@@ -1,19 +1,75 @@
...
+    public width: number,
+    public height: number,
+  ) {
+    if (width <= 0 || height <= 0) {
+      throw new Error('Length of sides must be greater than 0');
+    }
+
+    this.color = color;
+    this.height = height;
+    this.width = width;
```
Comment (etojeDenys) [inline src/figuresClasses.ts:64]: Remove this line as the width is already assigned in the constructor parameters

File: src/figuresClasses.ts [hunk 6]
```diff
@@ -1,19 +1,75 @@
...
+    public color: string,
+    public width: number,
+    public height: number,
+  ) {
+    if (width <= 0 || height <= 0) {
+      throw new Error('Length of sides must be greater than 0');
+    }
+
+    this.color = color;
+    this.height = height;
```
Comment (etojeDenys) [inline src/figuresClasses.ts:63]: Remove this line as the height is already assigned in the constructor parameters

File: src/figuresClasses.ts [hunk 7]
```diff
@@ -1,19 +1,75 @@
...
+  constructor(
+    public color: string,
+    public width: number,
+    public height: number,
+  ) {
+    if (width <= 0 || height <= 0) {
+      throw new Error('Length of sides must be greater than 0');
+    }
+
+    this.color = color;
```
Comment (etojeDenys) [inline src/figuresClasses.ts:62]: Remove this line as the color is already assigned in the constructor parameters

**Review by etojeDenys — CHANGES_REQUESTED** [review]

**Review by IvanFesenko — APPROVED** [review]

**Pull request closed without merging**